packages/model/addon/-private/model.js

```diff
@@ -579,6 +468,22 @@ class Model extends EmberObject {
     @private
     @type {Object}
   */
+  @tagged
+  get currentState() {
+    if (CUSTOM_MODEL_CLASS) {
+      return this.___recordState;
+    }
+    if (this.isDestroyed || this.isDestroying) {
```

Comment:
Curious, why do we need this?

Reply:
In the non CUSTOM_MODEL_CLASS case this is necessary for us to be able to hand the previous state to the user once the record is unloaded. This fixes an issue I noticed after the refactor to native model but was made evident by this PR. When `currentState` was a computed property it would return the cached value once the record was destroyed and not recompute.